zqd/space/manager.go

```diff
@@ -2,63 +2,57 @@ package space
 
 import (
 	"context"
+	"fmt"
 	"io/ioutil"
-	"os"
-	"path/filepath"
+	"path"
 	"sync"
 
+	"github.com/brimsec/zq/pkg/iosrc"
+	"github.com/brimsec/zq/pkg/s3io"
 	"github.com/brimsec/zq/zqd/api"
 	"github.com/brimsec/zq/zqd/storage"
 	"github.com/brimsec/zq/zqe"
 	"go.uber.org/zap"
 )
 
 type Manager struct {
-	rootPath string
+	rootPath iosrc.URI
 	spacesMu sync.Mutex
 	spaces   map[api.SpaceID]Space
 	names    map[string]api.SpaceID
 	logger   *zap.Logger
 }
 
-func NewManager(root string, logger *zap.Logger) (*Manager, error) {
+func NewManager(root iosrc.URI, logger *zap.Logger) (*Manager, error) {
 	mgr := &Manager{
 		rootPath: root,
 		spaces:   make(map[api.SpaceID]Space),
 		names:    make(map[string]api.SpaceID),
 		logger:   logger,
 	}
-
-	dirs, err := ioutil.ReadDir(root)
-	if err != nil {
-		return nil, err
+	var err error
+	var dirs []iosrc.URI
+	switch root.Scheme {
```

Comment:
What about `iosrc` supporting a way to list or walk through the objects of a directory-like uri?

Reply:
That's an interesting idea @alfred-landrum . Let me look into that.

Reply:
We already have the code somewhere in zar/zdx that does this.

Reply:
@alfred-landrum if it's alright with you I'm going to hold off on this for now. I agree that there should be some simple iosrc function here but I'm not quite sure what this should look like. Proposing we merge and revisit.